src/murmur/ServerUser.cpp

```diff
@@ -112,3 +112,53 @@ int BandwidthRecord::bandwidth() const {
 	return static_cast<int>((sum * 1000000ULL) / elapsed);
 }
 
+// Rate limiting: burst up to 30, 4 message per sec limit over longer time
+LeakyBucket::LeakyBucket() : tokensPerSec(4), maxTokens(30), currentTokens(0) {
```

Comment:
If you keep tokensPerSec and maxTokens constant then I don't see why these should be members. Maybe add them as parameters in the constructor too, so it can be set from for example the murmur.ini.

Or if you just don't want the ServerUser.h to be cluttered then move the LeakyBucket stuff to an extra header file.

Reply:
I set the values in the constructor in preparation for a pull request which adds the two parameters in the server configuration (`murmur.ini`).